src/pages/propose.astro

```diff
@@ -1,10 +1,151 @@
 ---
 import Layout from '../layouts/Layout.astro';
 import Card from '../components/Card.astro';
+
+export async function fetchContent({ request, response }) {
+  if (request.method === 'POST') {
+    try {
+      const data = await request.formData();
+      const name = data.get('name') as string;
+      const email = data.get('email') as string;
+      const topic = data.get('topic') as string;
+      const summary = data.get('summary') as string;
+
+      // Check if name has at least 6 characters
+      if (name && name.length < 6) {
+        throw new Error('Name must be at least 6 characters long');
+      }
+
+      // Check if email is valid (you can add more complex validation)
+      if (email && !isValidEmail(email)) {
+        throw new Error('Invalid email address');
+      }
+
+      // Check if topic and summary have at least 6 characters
+      if ((topic && topic.length < 6) || (summary && summary.length < 6)) {
+        throw new Error('Topic and summary must be at least 6 characters long');
+      }
+
+      // Retrieve all selected values for checkboxes
+      const proposalTypes = data.getAll('proposal-type');
+
+      // This is POST request. It's not going to be visible in browser.
+      // Check Network tab in Inspect Mode of Google Chrome or Firefox to see request.
+    } catch (error) {
+      if (error instanceof Error) {
+        console.error(error.message);
+      }
+    }
+  }
+}
+
+// Add a simple email validation function
+function isValidEmail(email: string | null): boolean {
+  // Just check if it contains '@'
+  return !!email && email.includes('@');
+}
 ---
 
-<Layout title="Tulsa Web Devs">
-  <main class="grid w-screen h-screen bg-white text-4xl place-items-center">
-    Proposal Features Here!
+<Layout title="propose">
+  <main>
+    <img
+      alt="Tulsa Web Devs logo"
+      src="/twd-logo.svg"
+      height="auto"
+      width="100%"
+    />
+
+    <div class="flex flex-col items-center m-2" id="header">
+      <h1 class="text-3xl max-md:text-2xl">Propose an idea!</h1>
+      <h2 class="text-2xl max-md:text-xl">
+        Have an idea?! Submit it! Just fill out this form!
+      </h2>
+    </div>
+
+    <form method="POST" id="proposalForm" class="m-4">
```

Comment:
I am happy to take a look at those api docs! When I click on the link, I get this:
![image](https://github.com/tulsawebdevs/website/assets/98660173/3c59e446-d4e4-486e-8a07-586faa592165)
I am still getting my feet wet in the GitHub game. Which am I supposed to click on? These seem like branches rather than readme files?
I was able to click on this, but the doc/ options in the dropdown seem to lead to the same "Contributing" readme.
![image](https://github.com/tulsawebdevs/website/assets/98660173/8cd102cf-78ce-42d1-beda-bdce9420b5d7)

Reply:
Happy to help! Unfortunately, I was not able to "commit suggestion". It gave me an error for some reason.

I was able to copy and paste this into the file, and commit it.

Reply:
Oops! That's my mistake. I was lazy and didn't add `https://` to my links, so GitHub treated them as relative links 🤦 

I'm updating the links now, here's a good one: [https://www.tulsawebdevs.org/docs/api](https://www.tulsawebdevs.org/docs/api)